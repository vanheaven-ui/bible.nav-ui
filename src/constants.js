export const BASE_URL = 'https://api.scripture.api.bible/v1/bibles/55212e3cf5d04d49-01';

export const BOOKS_URL = `${BASE_URL}/books?include-chapters=true&include-chapters-and-sections=true`;

export const RAPID_API_BASE = 'https://ajith-holy-bible.p.rapidapi.com/GetVerseOfaChapter';

const PARAMS = {
  notes: 'include-notes=false',
  titles: 'include-titles=false',
  chapterNum: 'include-chapter-numbers=true',
  verseNum: 'include-verse-numbers=true',
  verseSpan: 'include-verse-spans=false',
};

const CPARAMS = `?content-type=json&${PARAMS.notes}&${PARAMS.titles}&${PARAMS.chapterNum}&${PARAMS.verseNum}&${PARAMS.verseSpan}`;

const chaptersURL = id => `${BASE_URL}/chapters/${id}${CPARAMS}`;

export const dateOptions = {
  weekday: 'long', year: 'numeric', month: 'long', day: 'numeric',
};

export const VERSES_ARRAY = [
  { bookName: 'Jeremiah', chapterNum: '29', verseNum: '11' },
  { bookName: 'Psalms', chapterNum: '23', verseNum: '6' },
  { bookName: '1 Corinthians', chapterNum: '4', verseNum: '4' },
  { bookName: 'Phillipians', chapterNum: '4', verseNum: '13' },
  { bookName: 'John', chapterNum: '3', verseNum: '16' },
  { bookName: 'Romans', chapterNum: '8', verseNum: '28' },
  { bookName: 'Isaiah', chapterNum: '41', verseNum: '10' },
  { bookName: 'Psalms', chapterNum: '46', verseNum: '1' },
  { bookName: 'Galatians', chapterNum: '5', verseNum: '22' },
  { bookName: 'Hebrews', chapterNum: '11', verseNum: '1' },
  { bookName: '2 Timothy', chapterNum: '1', verseNum: '7' },
  { bookName: '1 Corinthians', chapterNum: '10', verseNum: '13' },
  { bookName: 'Proverbs', chapterNum: '22', verseNum: '6' },
  { bookName: 'Isaiah', chapterNum: '40', verseNum: '31' },
  { bookName: 'Joshua', chapterNum: '1', verseNum: '9' },
  { bookName: 'Hebrews', chapterNum: '12', verseNum: '2' },
  { bookName: 'Mathew', chapterNum: '11', verseNum: '28' },
  { bookName: 'Romans', chapterNum: '10', verseNum: '9' },
  { bookName: 'Phillipians', chapterNum: '2', verseNum: '3' },
  { bookName: 'Mathew', chapterNum: '5', verseNum: '45' },
  { bookName: 'Colossians', chapterNum: '3', verseNum: '1' },
  { bookName: 'John', chapterNum: '14', verseNum: '1' },
  { bookName: 'John', chapterNum: '14', verseNum: '14' },
  { bookName: 'Psalms', chapterNum: '91', verseNum: '2' },
  { bookName: 'Colossians', chapterNum: '3', verseNum: '23' },
  { bookName: 'James', chapterNum: '1', verseNum: '19' },
  { bookName: 'Psalms', chapterNum: '37', verseNum: '4' },
  { bookName: 'Psalms', chapterNum: '73', verseNum: '26' },
  { bookName: '1 Corinthians', chapterNum: '13', verseNum: '13' },
  { bookName: 'Psalms', chapterNum: '22', verseNum: '24' },
  { bookName: 'Isaiah', chapterNum: '41', verseNum: '13' },
  { bookName: 'Ecclesiastes', chapterNum: '7', verseNum: '9' },
];

export default chaptersURL;
