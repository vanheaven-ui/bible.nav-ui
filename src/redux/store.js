import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancer = composeWithDevTools(middlewareEnhancer);

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
