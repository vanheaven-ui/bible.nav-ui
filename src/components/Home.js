import { useEffect, useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { Link, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactLoading from 'react-loading';
import { BOOKS_URL, dateOptions } from '../constants';
import fetchData from '../services/fetchData';
import verseOfDay from '../services/verseOfDay';
import '../styles/home.css';
import persistLogin from '../util';
import retrieveChapters from '../redux/actions/chapters/chapters';
import getBookName from '../redux/actions/bookName';

const Home = ({ currentUser, login }) => {
  const hist = useHistory();
  const [isLoading, setIsLoading] = useState();
  const dispatch = useDispatch();
  const [VoD, setVoD] = useState('');
  const [ref, setRef] = useState('');

  // Variables for books of the bible
  const [newTestament, setNewTestament] = useState([]);
  const [oldTestament, setOldTestament] = useState([]);
  const [showOld, setShowOld] = useState(false);
  const [showNew, setShowNew] = useState(false);
  const [currentTime] = useState(new Date().toLocaleDateString('en', dateOptions));

  // variables to show chapters
  const [showChapters, setShowChapters] = useState(false);

  // local storage
  const favorites = JSON.parse(localStorage.getItem('favorites'));

  useEffect(() => {
    persistLogin(dispatch, currentUser, favorites, login);
    setIsLoading(true);
    verseOfDay().then(data => {
      setIsLoading(false);
      setVoD(data.Output);
      setRef(`${data.Book} ${data.Chapter}:${data.Verse}`);
    });

    fetchData(BOOKS_URL)
      .then(data => {
        setOldTestament(data.data.slice(0, 53));
        setNewTestament(data.data.slice(53));
      });
  }, []);

  const handleClick = e => {
    if (e.target.name === 'new' && !showNew) {
      setShowNew(true);
    } else {
      setShowNew(false);
    }

    if (e.target.name === 'old' && !showOld) {
      setShowOld(true);
    } else {
      setShowOld(false);
    }

    if (showChapters) {
      setShowChapters(false);
    }
  };

  // Function to get chapters of each book and dispatch them to store
  const fetchChapters = e => {
    dispatch(getBookName(e.target.name));
    retrieveChapters(dispatch, setShowChapters, hist, e.target.id);
  };

  return (
    <>
      <div className="App">
        <section className="hero">
          <p className="caret">
            <i className="fas fa-angle-double-down" />
          </p>
          <div className="hero-text">
            <Container>
              <Row>
                <Col id="left-col" xs={{ span: 12, order: 'last' }} md={{ span: 6, order: 'first' }}>
                  <div className="left-hero">
                    <h2> Navigating the Bible simplified</h2>
                    <p>Your favorite bible verse is a login away</p>
                    <button type="button" className="orange-button">
                      <Link to="/signup">START HERE</Link>
                    </button>
                  </div>
                </Col>
                <Col xs={{ span: 12, order: 'first' }} md={{ span: 6, order: 'last' }}>
                  <div className="hero-heading">
                    <small>{currentTime}</small>
                    <h4>VERSE OF THE DAY:</h4>
                    <h2>{ref}</h2>
                  </div>
                  <div className="verse-section">
                    { isLoading && <ReactLoading type="bubbles" color="#fff" width="100px" height="100px" /> }
                    { !isLoading && <p className="verse-of-day">{VoD}</p> }
                  </div>
                </Col>
              </Row>
            </Container>
          </div>
        </section>
        <section className="book-list">
          <h2>GET BOOKS OF THE BIBLE</h2>
          <Container fluid id="bible-sections">
            <Row>
              <Col xs={12} md={6} id="navigate">
                <h4>Let us navigate the sections of the bible</h4>
                <i className="fas fa-caret-right fa-3x right-fa" />
                <i className="fas fa-caret-down fa-3x down-fa" />
              </Col>
              <Col xs={12} md={6}>
                { !showOld && (
                <button
                  type="button"
                  name="new"
                  onClick={e => handleClick(e)}
                  title="Click me!"
                >
                  NEW TESTAMENT
                </button>
                )}
                { !showNew && (
                <button
                  type="button"
                  name="old"
                  onClick={e => handleClick(e)}
                  title="Click me!"
                >
                  OLD TESTAMENT
                </button>
                )}
                { showNew && (
                <div className="new">
                  { newTestament.length > 0 && newTestament.map(book => (
                    <button
                      name={book.name}
                      type="button"
                      id={book.id}
                      key={book.id}
                      onClick={e => fetchChapters(e)}
                    >
                      {book.name}
                    </button>
                  ))}
                </div>
                )}
                { showOld && (
                <div className="old">
                  { oldTestament.length > 0 && oldTestament.map(book => (
                    <button
                      name={book.name}
                      type="button"
                      id={book.id}
                      key={book.id}
                      onClick={e => fetchChapters(e)}
                    >
                      {book.name}
                    </button>
                  ))}
                </div>
                )}
              </Col>
            </Row>
          </Container>
        </section>
        <section className="favorites-pitch">
          <h3>CREATE YOUR FAVORITES</h3>
          <p>Have your favorites in one place by registering or signing in</p>
          <div className="buttons">
            <button type="button" className="orange-button"><Link to="/signup">REGISTER</Link></button>
            <button type="button" className="outline-btn"><Link to="/login">LOGIN</Link></button>
          </div>
        </section>
      </div>
    </>
  );
};

Home.propTypes = {
  currentUser: PropTypes.instanceOf(Object),
  login: PropTypes.func.isRequired,
};

Home.defaultProps = {
  currentUser: {},
};

export default Home;
