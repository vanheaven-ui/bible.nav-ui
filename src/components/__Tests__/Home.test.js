import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import store from '../../redux/store';
import '@testing-library/jest-dom';
import Home from '../Home';
import { dateOptions } from '../../constants';

describe('Home Component', () => {
  let dateMock = new Date();
  dateMock = jest.fn(() => 1620644512642);
  it('renders without crashing', () => {
    ReactDOM.render(
      <Provider store={store}><Router><Home /></Router></Provider>, rootDiv(),
    );
  });

  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <Provider store={store}><Router><Home /></Router></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct wrapper element', () => {
    const { container } = render(
      <Provider store={store}><Router><Home /></Router></Provider>,
    );
    expect(container.firstChild.classList.contains('App')).toBe(true);
  });
});
