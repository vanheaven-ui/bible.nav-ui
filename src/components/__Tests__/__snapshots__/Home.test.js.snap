// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component does not change unexpectedly 1`] = `
<div
  className="App"
>
  <section
    className="hero"
  >
    <p
      className="caret"
    >
      <i
        className="fas fa-angle-double-down"
      />
    </p>
    <div
      className="hero-text"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 col-12 order-md-first order-last"
            id="left-col"
          >
            <div
              className="left-hero"
            >
              <h2>
                 Navigating the Bible simplified
              </h2>
              <p>
                Your favorite bible verse is a login away
              </p>
              <button
                className="orange-button"
                type="button"
              >
                <a
                  href="/signup"
                  onClick={[Function]}
                >
                  START HERE
                </a>
              </button>
            </div>
          </div>
          <div
            className="col-md-6 col-12 order-md-last order-first"
          >
            <div
              className="hero-heading"
            >
              <small>
                Friday, May 14, 2021
              </small>
              <h4>
                VERSE OF THE DAY:
              </h4>
              <h2>
                
              </h2>
            </div>
            <div
              className="verse-section"
            >
              <p
                className="verse-of-day"
              >
                
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="book-list"
  >
    <h2>
      GET BOOKS OF THE BIBLE
    </h2>
    <div
      className="container-fluid"
      id="bible-sections"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-12"
          id="navigate"
        >
          <h4>
            Let us navigate the sections of the bible
          </h4>
          <i
            className="fas fa-caret-right fa-3x right-fa"
          />
          <i
            className="fas fa-caret-down fa-3x down-fa"
          />
        </div>
        <div
          className="col-md-6 col-12"
        >
          <button
            name="new"
            onClick={[Function]}
            title="Click me!"
            type="button"
          >
            NEW TESTAMENT
          </button>
          <button
            name="old"
            onClick={[Function]}
            title="Click me!"
            type="button"
          >
            OLD TESTAMENT
          </button>
        </div>
      </div>
    </div>
  </section>
  <section
    className="favorites-pitch"
  >
    <h3>
      CREATE YOUR FAVORITES
    </h3>
    <p>
      Have your favorites in one place by registering or signing in
    </p>
    <div
      className="buttons"
    >
      <button
        className="orange-button"
        type="button"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          REGISTER
        </a>
      </button>
      <button
        className="outline-btn"
        type="button"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          LOGIN
        </a>
      </button>
    </div>
  </section>
</div>
`;
